init/setup-rwdev.sh

```diff
@@ -7,17 +7,12 @@
 set -e
 
 dev=/dev/xvdb
-max_size=1073741824  # check at most 1 GiB
 
 if [ -e "$dev" ] ; then
     # The private /dev/xvdb device is present.
 
-    # check if private.img (xvdb) is empty - all zeros
-    private_size=$(( $(blockdev --getsz "$dev") * 512))
-    if [ $private_size -gt $max_size ]; then
-        private_size=$max_size
-    fi
-    if cmp --bytes $private_size "$dev" /dev/zero >/dev/null && { blkid -p "$dev" >/dev/null; [ $? -eq 2 ]; }; then
+    # check if private.img (xvdb) is empty
+    if { blkid -p "$dev" >/dev/null; [ $? -eq 2 ]; }; then
```

Comment:
This seems potentially dangerous.

Reply:
In practice we are looking for a filesystem we create ourselves, and ext4 is perfectly detected (`-p` makes blkid really read the device, not use cache). But also, blkid detects tens of filesystems, including quite exotic ones. Do you have some specific scenario in mind that would fail this? Perhaps compare first 10MB  to /dev/zero with an assumption that any kind of FS signature would be there, even if not recognized by blkid? 

Reading first 1GB on every DispVM startup is something I'd really like to avoid, as those need to start fast, ideally under 1s (we're quite far from there). But also speeding up fresh VM startup is desirable.

Reply:
Comparing the first 10MB is fine.